### PR TITLE
Add initial support for Markdown

### LINKED ISSUE #7: Add Markdown support
I would like to contribute to your project to allow the program to convert .md files to .html.
I would like to make the following code contributions:

1. Modify the file handling so that it supports both .txt files as well as .md (i.e., Markdown files). If the file extension is .txt, process the file exactly the same way it is done now (i.e., no changes). However, if the file extension is .md, add some extra processing (a new function, another class, a new module, etc. as makes sense) to parse one aspect of Markdown into HTML.
2. Allow links in .md syntax to be converted to links in .html syntax.

If you would like, please give feedback about how you want it done.

### PR DESCRIPTION
**Pull request by kliu57**
Markdown files are accepted as input, and html links in Markdown links are converted to html links. Closes #7

### CODE REVIEW

File: src/ConvertTxtToHtml.java [hunk 1]
```diff
@@ -98,20 +101,20 @@ private static void processFile(File inputFile, String outputPath) throws IOExce
         htmlContent.append("</head>\n<body>\n");
 
         if (hasTitle) {
```
Comment (WangGithub0) [inline src/ConvertTxtToHtml.java:103]: could you only use convertLinks for md file?

File: src/ConvertTxtToHtml.java [hunk 2]
```diff
@@ -98,20 +101,20 @@ private static void processFile(File inputFile, String outputPath) throws IOExce
...
-            htmlContent.append("<h1>").append(title).append("</h1>\n");
+            htmlContent.append("<h1>").append(convertLinks(title)).append("</h1>\n");   // convert links in the title if any
```
Comment (WangGithub0) [inline src/ConvertTxtToHtml.java]: could you using convertLinks in a try{}

**Comment by WangGithub0** [comment]
I try this 
image
and get
image

could you fix it? thanks

File: src/ConvertTxtToHtml.java (hunk 1 above)
Comment (kliu57) [reply src/ConvertTxtToHtml.java:103]: I made the fix so that convertLinks will only be run for md files

**Comment by kliu57** [comment]
> I try this image and get image
> 
> could you fix it? thanks

I fixed this bug by updating the regex used to search for links

**Comment by WangGithub0** [comment]
LGTM!

**Pull request merged**